docs/copilot/workspace-context.md

```diff
@@ -0,0 +1,72 @@
+---
+Order: 7
+Area: copilot
+TOCTitle: Workspace Context
+ContentId: c77dcce9-4ba9-40ac-8ae5-2df855088090
+PageTitle: Chat using @workspace Context References
+DateApproved: 02/28/2024
+MetaDescription: Overview of how the @workspace Copilot chat participants works and how it manages context.
+---
+# How the @workspace Copilot chat participant works
+
+Referencing `@workspace` in Copilot Chat lets you ask questions about your entire codebase. Copilot smartly retrieves relevant files and symbols based on your question, which it then references in its answer. Using `@workspace` references, Copilot Chat can provide in-depth expert help for tasks like:
+
+- Finding existing code in your codebase:
+  - `"@workspace where is database connecting string configured?"`
+  - `"@workspace is there a library for date validation?"`
+  - `"Where are tests defined?"`
+- Making plans for larger codebase edits:
+  - `"@workspace how can I add a hover text to a button?"`
+  - `"@workspace add date validation to #selection"`
+  - `"@workspace how do I add a new a new API route?"`
+- Explaining higher-level concepts in a codebase:
+  - `"@workspace how is authentication implemented?"`
+  - `"@workspace which API routes does this service provide?"`
+  - `"How do I build this project?"`
+
+## What context does @workspace have?
+
+The goal of the `@workspace` is to provide answers to questions about your full workspace that you have open in VS Code. It indexes different sources of information and applies different search strategies to come up with the most relevant context:
+
+- All files in the VS Code workspace, except for files that are included in `.gitignore` files
+- Workspace structure, such as folder and file names
+- GitHub's code search index, if the workspace is a GitHub repository and indexed by code search
+- Symbols and definitions in the workspace
+- Currently selected text (`#selection`) or visible text in the active editor (`#editor`)
```

Comment:
Do we have docs on `#variables`? It might be useful to link to some docs on them and/or explain a bit more about them here in case folks haven't seen this syntax.

Reply:
Yes, these are explained in the updated Copilot docs (#7090 )